### PR TITLE
[WinForms] Fix UIA event on node insertions into a `TreeView` object

### PR DESCRIPTION
**Pull request by NVoronchev**
The UIA event doesn't emited by means of insertion a new `TreeNode` to a `TreeNodeCollection` of the respective `TreeNode`.

### CODE REVIEW

**Comment by akoeplinger** [comment]
@monojenkins build

File: mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs [hunk 1]
```diff
@@ -90,8 +90,11 @@ internal TreeNodeCollection (TreeNode owner)
 			set {
 				if (index < 0 || index >= Count)
 					throw new ArgumentOutOfRangeException ("index");
-				SetupNode (value);
+				
+				OnUIACollectionChanged (CollectionChangeAction.Remove, nodes [index]);
 				nodes [index] = value;
```
Comment (filipnavara) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs]: The order of operations is reversed here. While the UIA API is not well documented this seems like a breaking change. 

```csharp
var removed = nodes [index];
nodes [index] = value;
OnUIACollectionChanged (CollectionChangeAction.Remove, removed);
```

Comment (NVoronchev) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs]: @filipnavara, thank you! You are right, one should raise "remove"- event after the node is removed. How do you like this approach:
```
    set {
        if (index < 0 || index >= Count)
            throw new ArgumentOutOfRangeException ("index");
        
-       OnUIACollectionChanged (CollectionChangeAction.Remove, nodes [index]);
-       nodes [index] = value;
-       
-       SetupNode (value);
+       RemoveAt (index, false);
+       Insert (index, value);
    }
```

Comment (filipnavara) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs]: That will cause unnecessary array resizing which could be costly. I'd go with the less disruptive change.

Comment (NVoronchev) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/TreeNodeCollection.cs]: Please, find the PR update

**Comment by akoeplinger** [comment]
@monojenkins build

**Pull request merged**